core/builtin.py

```diff
@@ -484,9 +484,8 @@ def Cd(argv, mem):
 
 
 def PrintDirStack(dir_stack):
-  if dir_stack:
-    print(' '.join(dir_stack))
-
+  print(os.getcwd() + ((' ' + ' '.join(dir_stack)) if dir_stack else ''))
```

Comment:
This feels a bit awkward.  I think `if dir_stack:` is a lot more straightforward.

But see the other comment about unifying this calculation.